plugins/inventory/vmware_vm_inventory.py

```diff
@@ -375,142 +497,141 @@ def parse(self, inventory, loader, path, cache=True):
-                if not vm_obj.obj.config:
-                    # Sometime orphaned VMs return no configurations
-                    continue
-
-                current_host = vm_obj_property.val + "_" + vm_obj.obj.config.uuid
-
-                if current_host not in hostvars:
-                    hostvars[current_host] = {}
-                    self.inventory.add_host(current_host)
-
-                    host_ip = vm_obj.obj.guest.ipAddress
-                    if host_ip:
-                        self.inventory.set_variable(current_host, 'ansible_host', host_ip)
-
-                    self._populate_host_properties(vm_obj, current_host)
-
-                    # Only gather facts related to tag if vCloud and vSphere is installed.
-                    if HAS_VSPHERE and self.pyv.with_tags:
-                        # Add virtual machine to appropriate tag group
-                        vm_mo_id = vm_obj.obj._GetMoId()
-                        vm_dynamic_id = DynamicID(type='VirtualMachine', id=vm_mo_id)
-                        attached_tags = tag_association.list_attached_tags(vm_dynamic_id)
-
-                        for tag_id in attached_tags:
-                            self.inventory.add_child(tags_info[tag_id], current_host)
-                            cacheable_results[tags_info[tag_id]]['hosts'].append(current_host)
-
-                    # Based on power state of virtual machine
-                    vm_power = str(vm_obj.obj.summary.runtime.powerState)
-                    if vm_power not in cacheable_results:
-                        cacheable_results[vm_power] = {'hosts': []}
-                        self.inventory.add_group(vm_power)
-                    cacheable_results[vm_power]['hosts'].append(current_host)
-                    self.inventory.add_child(vm_power, current_host)
-
-                    # Based on guest id
-                    vm_guest_id = vm_obj.obj.config.guestId
-                    if vm_guest_id and vm_guest_id not in cacheable_results:
-                        cacheable_results[vm_guest_id] = {'hosts': []}
-                        self.inventory.add_group(vm_guest_id)
-                    cacheable_results[vm_guest_id]['hosts'].append(current_host)
-                    self.inventory.add_child(vm_guest_id, current_host)
-
-        for host in hostvars:
-            h = self.inventory.get_host(host)
-            cacheable_results['_meta']['hostvars'][h.name] = h.vars
-
-        return cacheable_results
-
-    def _populate_host_properties(self, vm_obj, current_host):
+                properties[vm_obj_property.name] = vm_obj_property.val
+                # Custom values
+                if 'customValue' in vm_properties:
+                    field_mgr = self.pyv.content.customFieldsManager.field
+                    for cust_value in vm_obj.obj.customValue:
+                        properties[[y.name for y in field_mgr if y.key == cust_value.key][0]] = cust_value.value
```

Comment:
```suggestion
            # Custom values
            if 'customValue' in vm_properties:
                field_mgr = self.pyv.content.customFieldsManager.field
                for cust_value in vm_obj.obj.customValue:
                    properties[[y.name for y in field_mgr if y.key == cust_value.key][0]] = cust_value.value
```
Move it outside of the vm properties loop.

Reply:
Done.